karabo_data/tests/mockdata/detectors.py

```diff
@@ -17,22 +17,37 @@ class DetectorModule:
         'trailer',
     ]
 
-    def __init__(self, device_id, frames_per_train=64):
+    def __init__(self, device_id, frames_per_train=64, raw=True):
         self.device_id = device_id
         self.frames_per_train = frames_per_train
+        if not raw:
+            # Drop the first dim (gain) if not raw (proc data)
+            self.image_dims = self.image_dims[1:]
+        self.raw = raw
 
     def write_control(self, f):
         """Write the CONTROL and RUN data, and the relevant parts of INDEX"""
         pass
 
     @property
     def image_keys(self):
-        return [
-            ('cellId', 'u2', (1,)),
-            ('data', 'u2', self.image_dims),
-            ('length', 'u4', (1,)),
-            ('status', 'u2', (1,)),
-        ]
+        if self.raw:
+            return [
+                ('cellId', 'u2', (1,)),
+                ('data', 'u2', self.image_dims),
+                ('length', 'u4', (1,)),
+                ('status', 'u2', (1,)),
+            ]
+
+        else:
+            return [
+                ('cellId', 'u2', (1,)),
+                ('data', 'f4', self.image_dims),
+                ('mask', 'u2', self.image_dims),
+                ('gain', 'u2', self.image_dims),
```

Comment:
In an example I'm looking at (`/gpfs/exfel/exp/SPB/201701/p002012/raw/r0243`), mask and gain are uint8, which is `'u1'` in this system (1 byte). Do you have an example where they're 2 bytes?

Reply:
I just checked one of the more recent (201930) calibrated runs and the mask is u4 whereas the gain is u1. I'll fix that